base/heartbeat.go

```diff
@@ -65,8 +65,9 @@ type HeartbeatListener interface {
 //       e.g  Default for a 4 node cluster: 12 ops/second
 type couchbaseHeartBeater struct {
 	bucket                  Bucket
-	nodeUUID                string
 	keyPrefix               string
+	groupID                 *string
```

Comment:
*string is a relatively uncommon pattern in go, and I'm not sure it completely makes sense here.  We don't actually want to support cases where the string is changed once it's been passed to the heartbeater.  I don't think we support a configGroupID of empty string (do we?), so I'd prefer to just work with `string`.